plugins/processors/k8sdecorator/k8sdecorator_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package k8sdecorator
+
+import (
+	. "github.com/aws/amazon-cloudwatch-agent/internal/containerinsightscommon"
+	"github.com/aws/amazon-cloudwatch-agent/internal/structuredlogscommon"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func TestDisableMetricExtraction(t *testing.T) {
+	tags := map[string]string{MetricType: TypeCluster}
+	fields := map[string]interface{}{MetricName(TypeCluster, NodeCount): 10, MetricName(TypeCluster, FailedNodeCount): 1}
+
+	m1 := metric.New("testClusterMetric", tags, fields, time.Now())
```

Comment:
Another thing I would consider would be using [Given - When - Then](https://en.wikipedia.org/wiki/Given-When-Then) rule for unit test: to make it easier to understand the work flow since `attributes in Fields` is a hacky way of us passing logs events through pipeline.

Reply:
Updated as suggested!